python/lsst/sims/maf/metrics/calibrationMetrics.py

```diff
@@ -161,6 +163,174 @@ def run(self, dataslice, slicePoint=None):
+                 mjdCol='expMJD', filterCol='filter', seeingCol='FWHMgeom',
+                 rmag=20., SedTemplate='flat', badval=-666,
+                 atm_err=0.01, thetaRange=0., snrLimit=5, **kwargs):
+        """
+        instantiate metric
+
+        m5Col = column name for inidivual visit m5
+        mjdCol = column name for exposure time dates
+        filterCol = column name for filter
+        seeingCol = column name for seeing (assumed FWHM)
+        rmag = mag of fiducial star in r filter.  Other filters are scaled using sedTemplate keyword
+        sedTemplate = template to use (can be 'flat' or 'O','B','A','F','G','K','M')
+        atm_err = centroiding error due to atmosphere in arcsec
+        thetaRange = range of parallax offset angles to demand (in radians) default=0 means no range requirement
+        snrLimit = only include points above the snrLimit (default 5) when computing thetaRange.
+        """
+
+        cols = ['ra_pi_amp', 'dec_pi_amp', m5Col, mjdCol, filterCol, seeingCol]
+        units = 'ratio'
+        super(ParallaxCoverageMetric, self).__init__(cols,
+                                                     metricName=metricName, units=units,
+                                                      **kwargs)
+        self.m5Col = m5Col
+        self.seeingCol = seeingCol
+        self.filterCol = filterCol
+        self.mjdCol = mjdCol
+
+        # Demand the range of theta values
+        self.thetaRange = thetaRange
+        self.snrLimit = snrLimit
+
+        filters=['u','g','r','i','z','y']
+        self.mags={}
+        if SedTemplate == 'flat':
+            for f in filters:
+                self.mags[f] = rmag
+        else:
+            self.mags = utils.stellarMags(SedTemplate, rmag=rmag)
+        self.atm_err = atm_err
+
+    def _thetaCheck(self, ra_pi_amp, dec_pi_amp, snr):
+        good = np.where(snr >= self.snrLimit)
+        theta = np.arctan2(dec_pi_amp[good], ra_pi_amp[good])
+        # Make values between 0 and 2pi
+        theta = theta-np.min(theta)
+        result = 0.
+        if np.max(theta) >= self.thetaRange:
+            # Check that things are in differnet quadrants
+            theta = (theta+np.pi) % 2.*np.pi
+            theta = theta-np.min(theta)
+            if np.max(theta) >= self.thetaRange:
+                result = 1
+        return result
+
+    def computeWeights(self, dataSlice, snr):
+        # Compute centroid uncertainty in each visit
+        position_errors = np.sqrt(astrom_precision(dataSlice[self.seeingCol], snr)**2+self.atm_err**2)
+        weights = 1./position_errors**2
+        return weights
+
+    def weightedR(self, dec_pi_amp, ra_pi_amp, weights):
+        ycoord = dec_pi_amp-np.average(dec_pi_amp, weights=weights)
+        xcoord = ra_pi_amp-np.average(ra_pi_amp,weights=weights)
+        radius = np.sqrt(xcoord**2+ycoord**2)
+        aveRad = np.average(radius, weights=weights)
+        return aveRad
```

Comment:
Should the above two methods be private methods? would they be called outside of the (public) run method below?